### PR TITLE
fix: invaders late join issues

### LINKED ISSUE #28: enemies not showing up when joining an inprogress game
**Describe the bug**
Joining a game already playing in a session results in the enemies not showing up. Bullets rain down still, however.

**To Reproduce**
1. Build the game and run twice.
2. Have first copy of game host
3. second copy joins localhost game
4. run a third copy of the game
5. have third copy join localhost

**Expected behavior**
I would expect the game to be fully recreated for the late joiner, allowing them to see all the graphical elements the other two clients see.

**Environment (please complete the following information):**
 - OS: Windows 10
 - Unity Version: 2020.3.12f1
 - MLAPI Version: 0.1.0
 - Bitesize Samples Repository Version/Commit https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/commit/3dc3c39f8a9e5de91c544fefd78188cc739143d5

### PR DESCRIPTION
**Pull request by LPLafontaineB**
This PR fixes how player color is set so that it is set for late-joining players too. It also simplifies how the scene transition callbacks are setup and adds an RPC call to replicate the timer for late-joining players.

Fixes #28

### CODE REVIEW

**Review by SamuelBellomo — APPROVED** [review]

File: Basic/Invaders/Assets/Scripts/SceneTransitionHandler.cs [hunk 1]
```diff
@@ -106,18 +113,12 @@ public void SwitchScene(string scenename)
         }
     }
 
-    void OnLoadComplete(ulong clientId, string sceneName, LoadSceneMode loadSceneMode)
+    private void OnLoadComplete(ulong clientId, string sceneName, LoadSceneMode loadSceneMode)
```
Comment (fernando-cortez) [inline Basic/Invaders/Assets/Scripts/SceneTransitionHandler.cs:116]: Nitpick: private.

File: Basic/Invaders/Assets/Scripts/InvadersGame.cs [hunk 2]
```diff
@@ -147,6 +147,8 @@ public override void OnNetworkDespawn()
             m_Enemies.Clear();
             m_Shields.Clear();
         }
+
+        NetworkManager.OnClientConnectedCallback -= OnClientConnected;
```
Comment (fernando-cortez) [inline Basic/Invaders/Assets/Scripts/InvadersGame.cs:151]: Just a comment to use NetworkManager.Singleton here and below for parity in the codebase.

**Review by fernando-cortez — APPROVED** [review]
Besides nitpicks, gtg.

File: Basic/Invaders/Assets/Scripts/SceneTransitionHandler.cs (hunk 1 above)
Comment (LPLafontaineB) [reply Basic/Invaders/Assets/Scripts/SceneTransitionHandler.cs:116]: I added it here to be consistent with the rest of the project

**Review by fernando-cortez — APPROVED** [review]

**Pull request merged**